src/Features/Core/Portable/MoveToNamespace/AbstractMoveToNamespaceService.cs

```diff
@@ -153,17 +169,35 @@ private static bool ContainsMultipleTypesInSpine(SyntaxNode node)
             }
         }
 
+        private static async Task<ImmutableArray<ISymbol>> GetMemberSymbolsAsync(Document document, SyntaxNode container, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            switch (container)
+            {
+                case TNamespaceDeclarationSyntax namespaceNode:
+                    var containerSymbol = (INamespaceSymbol)semanticModel.GetDeclaredSymbol(container);
+                    return containerSymbol.GetMembers().SelectAsArray(m => (ISymbol)m);
+
+                case TCompilationUnitSyntax compilationUnit:
+                    var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+                    var members = syntaxFacts.GetMembersOfCompilationUnit(compilationUnit);
+                    Debug.Assert(members.Count == 1);
+                    return members.SelectAsArray(member => semanticModel.GetDeclaredSymbol(member));
+
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(container);
+            }
+        }
+
         private static async Task<MoveToNamespaceResult> MoveItemsInNamespaceAsync(
             Document document,
             SyntaxNode container,
             string targetNamespace,
             CancellationToken cancellationToken)
         {
-            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            var containerSymbol = (INamespaceSymbol)semanticModel.GetDeclaredSymbol(container);
-            var members = containerSymbol.GetMembers();
-            var newNameOriginalSymbolMapping = members
-                .ToImmutableDictionary(symbol => GetNewSymbolName(symbol, targetNamespace), symbol => (ISymbol)symbol);
+            var memberSymbols = await GetMemberSymbolsAsync(document, container, cancellationToken).ConfigureAwait(false);
+            var newNameOriginalSymbolMapping = memberSymbols
+                .ToImmutableDictionary(symbol => GetNewSymbolName(symbol, targetNamespace), symbol => symbol);
```

Comment:
1. any chance that GetNewSymbolName could cause collisions? 
2. what happens in collisions?  does ToImmutableDictionary throw, or does it overwrite?

Reply:
I don't think that's possible, otherwise it'd mean multiple members in a namespace decl have identical name.

Reply:
And `ToImmutableDictionary` throws if two element has the same key

Reply:
hmm, what if we have multiple decls of same partial type in the file... @ryzngard did you try that before?

Reply:
I have added calls to `IChangeNamespaceService.CanChangeNamespaceAsync` before offering any code actions.